k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -1202,6 +1257,28 @@ public void expunge() throws MessagingException {
         }
     }
 
+    public List<String> expungeUsingQresync() throws MessagingException {
+        open(OPEN_MODE_RW);
+        checkOpen();
+
+        try {
+            List<ImapResponse> expungeResponses = executeSimpleCommand("EXPUNGE");
+            handleExpungeResponses(expungeResponses);
+            return ImapUtility.extractVanishedUids(expungeResponses);
+        } catch (IOException ioe) {
+            throw ioExceptionHandler(connection, ioe);
+        }
+    }
+
+    private void handleExpungeResponses(List<ImapResponse> imapResponses) {
```

Comment:
This doesn't really 'handle' the response - it just searches them for a higher mod seq.

`updateHighestModSequenceFromResponses`.